packages/nimble-components/src/table/index.ts

```diff
@@ -66,15 +97,39 @@ export class Table<
 
         // Ignore any updates that occur prior to the TanStack table being initialized.
         if (this.tableInitialized) {
-            this.updateTableOptions({ data: this.data });
+            this.trySetData(this.data);
```

Comment:
Say for example the user has an array assigned to data, they mutate the data and assign the same array again (to trigger an update). Would we need the shallow copy like the idFieldNameChange above for tanstack to detect?

Maybe after validation, trySetData should always create a shallow copy to pass to updateTableOptions?

```ts
const arr = []; // good data
table.data = arr
table.idFieldName = '' // good id
// sometime later
arr[0] //mutate but still good
table.data = arr // do assignment to trigger update
```

Reply:
Currently, `data` is an `@observable` property on the table and the code to update TanStack happens within `dataChanged`. I'm not aware of a scenario where FAST would detect a change to the array but TanStack would not.

Perhaps it is better to create a setter for `data` rather than have it be observable, particularly because the template no longer binds to `data` (now it uses `tableData`). I'll play around with this, and assuming a setter gives us better behavior, I'll post a follow-up PR to change the approach.

Reply:
> I'm not aware of a scenario where FAST would detect a change to the array but TanStack would not.

I think then I don't fully understand why the shallow copy is needed for calling trySetData in idFieldNameChanged but not for calling trySetData in dataChanged.

My mental model is that calling updateTableOptions internally was just checking if the data reference was pointing to the same object or something and creating a shallow copy kicked it into recognizing a change by pointing to a new one. If that's the case then I would imagine we need a shallow copy in both cases.

Just trying to make sure it's clear when we need a shallow copy and maybe the comment could clarify a bit when it's needed.

Reply:
The reason that a shallow copy isn't needed in the code path where the data changes is because the current implementation requires the client to do that shallow copy. Otherwise, FAST won't detect any changes to the array, and our code to update TanStack won't get hit at all. With the current table code in master (unchanged with this PR), the last line in your code snippet above won't actually trigger an update. The client would need to write `table.data = [ ...arr ]` and do the shallow copy themselves.

My proposal is to (in a separate PR) make the `data` property on the table have a getter/setter rather than rely on FAST's change detection on the array. With that change, we will need to do the shallow copy both when the data is set and when the id changes, and the client will no longer be required to do so.

Reply:
perfect, thanks for the clarification! 🚢-it!

Reply:
I created #958 so we wouldn't lose track of needing to make this change.